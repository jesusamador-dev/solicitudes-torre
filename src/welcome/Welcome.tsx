/* eslint-disable jsx-a11y/anchor-is-valid */
import React, {Suspense} from 'react';
import '../assets/css/welcome/welcome.css';
const CoreStyles = React.lazy(()=>import('mf_react_core/CoreStyles'));
import SafeView from '../safe/SafeView';

const Welcome = () => {
  return (
    <SafeView
      color='white'
      title={`${process.env.REACT_APP_MF_NAME}
              ${process.env.REACT_APP_MF_VERSION}
              ${process.env.REACT_APP_ENVIRONMENT}`}
    >
      <Suspense fallback={<></>}>
        <CoreStyles>
          <div className=''>
            <header className='mf-name '>
              <div className=''>
                <div className=' mf-c-center mf-font'>
                  <div>{process.env.REACT_APP_MF_NAME}</div>
                  <div>{process.env.REACT_APP_MF_VERSION}</div>
                  <div>{process.env.REACT_APP_ENVIRONMENT}</div>
                </div>
              </div>
            </header>
            <div className='mf-bottom'></div>
          </div>
        </CoreStyles>
      </Suspense>
    </SafeView>
  );
};

export default Welcome;
