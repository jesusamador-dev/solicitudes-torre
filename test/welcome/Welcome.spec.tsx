import React from 'react';
import Welcome from '../../src/welcome/Welcome';
import renderer from 'react-test-renderer';

const LazyComp = React.lazy(() => import('../mock/LazyComp'));

jest.mock('@/skeleton/Skeleton',() => ({children, ...props}) => <div data-testid="ds-skeleton" {...props} data-circle={true} data-height={10} data-width={20}>{children}</div>)
jest.mock('@/safe/SafeView',() => ({children, ...props}) => <div data-testid="ds-safe" {...props}>{children}</div>)

describe('np test', function () {
  it('is testing a sample Welcome.', async () => {
    const tree = renderer
      .create(
        <>
          <Welcome />
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

