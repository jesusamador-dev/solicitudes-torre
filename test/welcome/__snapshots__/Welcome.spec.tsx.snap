// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Skeleton test is testing a sample skeleton. 1`] = `
Array [
  <span
    aria-busy="true"
    aria-live="polite"
    className=""
  >
    <span
      className="AurumSkeleton"
      style={
        Object {
          "borderRadius": "0%",
          "height": "20px",
          "width": "100%",
        }
      }
    >
      ‌
    </span>
    <br />
  </span>,
  <span
    aria-busy="true"
    aria-live="polite"
    className="a-class"
  >
    <span
      className="AurumSkeleton"
      style={
        Object {
          "borderRadius": "12px",
          "height": "10px",
          "width": "10px",
        }
      }
    >
      ‌
    </span>
    <br />
  </span>,
  <span
    aria-busy="true"
    aria-live="polite"
    className="a-class"
  >
    <span
      className="AurumSkeleton"
      style={
        Object {
          "borderRadius": "12px",
          "height": "100px",
          "width": "100px",
        }
      }
    >
      ‌
    </span>
    <br />
  </span>,
  <span
    aria-busy="true"
    aria-live="polite"
    className="a-class"
  >
    <span
      className="AurumSkeleton"
      style={
        Object {
          "borderRadius": "0%",
          "height": "10px",
          "width": "10%",
        }
      }
    >
      ‌
    </span>
    <br />
  </span>,
  <span
    aria-busy="true"
    aria-live="polite"
    className="a-class"
  >
    <span
      className="AurumSkeleton"
      style={
        Object {
          "borderRadius": "45px",
          "height": "10px",
          "width": "10%",
        }
      }
    >
      ‌
    </span>
    <br />
  </span>,
]
`;

exports[`Welcome safeview test is testing a sample safeview. 1`] = `"loading"`;

exports[`Welcome test is testing a sample Welcome. 1`] = `null`;
